operator/pkg/controllers/etcd/etcd.go

```diff
@@ -115,59 +108,17 @@ func (e *Provider) createService(ctx context.Context, controlPlane *v1alpha1.Con
 	return nil
 }
 
-func (e *Provider) createETCDCerts(ctx context.Context, controlPlane *v1alpha1.ControlPlane) error {
-	// create the ETCD ROOT CA certs and key
-	caCert, caKey, err := e.createSecret(ctx, controlPlane, etcdRootCACertConfig(controlPlane))
-	if err != nil {
-		return fmt.Errorf("creating root CA for name %v, %w", controlPlane.ClusterName(), err)
-	}
-	// create etcd server, peer, healthcheck and etcdAPIClient certs and key
-	for _, certConfig := range certListFor(controlPlane, caCert, caKey) {
-		if _, _, err := e.createSecret(ctx, controlPlane, certConfig); err != nil {
-			return fmt.Errorf("creating certs and key name %s, %w,", certConfig.Name, err)
-		}
+func (p *Provider) createETCDSecrets(ctx context.Context, controlPlane *v1alpha1.ControlPlane) error {
+	// create the root CA, certs and key for etcd
+	if err := secrets.WithRootCAName(p.kubeClient,
```

Comment:
I'd avoid passing dependencies like p.kubeClient as a parameter and instead inject it at object initialization time as a private member.